File: 编译原理/TinyScript-Js/src/parser/ast/AssignStmt.js
const ASTNodeTypes = require('./ASTNodeTypes')
const Stmt = require('./Stmt')

/**
 * 抽象语法树 - 赋值语句节点
 */
class AssignStmt extends Stmt {
  /**
   * @param {ASTNode} _parent
   * @param {ASTNodeTypes} _type
   * @param {string} _label
   */
  constructor(_parent) {
    super(_parent, ASTNodeTypes.ASSIGN_STMt, 'assign')
  }
}

module.exports = AssignStmt

File: 编译原理/TinyScript/src/main/java/parser/ast/AssignStmt.java
package parser.ast;

/**
 * 抽象语法树 - 赋值语句节点
 */
public class AssignStmt extends Stmt {
    public AssignStmt(ASTNode _parent) {
        super(_parent, ASTNodeTypes.ASSIGN_STMT, "assign");
    }
}
